src/qt/transactionview.cpp

```diff
@@ -443,6 +444,15 @@ void TransactionView::bumpFee()
     }
 }
 
+void TransactionView::bumpFee()
+{
+    try {
+        bumpFeeHelper();
+    } catch (const NonFatalCheckError& e) {
+        Q_EMIT nonFatalCheckError(QString::fromStdString(e.what()));
+    }
```

Comment:
No reason to tie this to to the CHECK_NONFATAL implementation or to have exceptions pass through QT code which by design does not use them. Would change this to not be tied to a specific exception type:

```
try {
    bumpFeeHelper();
} catch (const std::exception& e) {
    Q_EMIT runawayException(QString::fromStdString(e.what()));
} catch (...) {
    Q_EMIT runawayException("Unknown failure occurred");
}
```

It looks like other code could stay the same except to renaming `nonFatalCheckError` to `runawayException`

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18897#issuecomment-624710077).